### PR TITLE
Feat: Added json option to output json data

### LINKED ISSUE #186: Show files in webtorrent info
**What version of this package are you using?**
3.5.3

**What problem do you want to solve?**
I maintain an mpv script for direct integration with mpv (various advantages over --mpv) and would like to be able to simply get a list of urls that webtorrent creates. Currently, in order to do this I have to parse the html of `localhost:<port>` which isn't ideal.

**What do you think is the correct solution to this problem?**
I previously suggested #132. As an alternative, it would be nice if `webtorrent info` provided a list of files inside the torrent. I could then concatenate `localhost:<port>/<#>/` to the filenames to get the urls. I think this would be useful besides for my limited use case.

Other commandline tools can do this for a torrent file, but I'm not aware of any that can format the file list for scripts (e.g. in json), and I'd rather only have webtorrent-cli as a dependency of the script. This is not a huge deal, but I think it would be a nice improvement

**Are you willing to submit a pull request to implement this change?**
I don't have the time/knowledge.

### PR DESCRIPTION
**Pull request by OsmanAltun**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Documentation update
[ ] Bug fix
[x] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
added `--json` option which prints a json object containing:
 - href (base server URL)
 - argv (parsed arguments)
 - files (file names in array)

if u want specific file u can do `href/index/file_name`, make sure `file_name` is encoded
**Which issue (if any) does this pull request address?**
 - resolves #186 

**Is there anything you'd like reviewers to focus on?**
Anything I should add/remove from the JSON object?

### CODE REVIEW

File: bin/cmd.js [hunk 1]
```diff
@@ -111,7 +112,7 @@ process.on('SIGINT', gracefulExit)
 process.on('SIGTERM', gracefulExit)
 
 yargs
-  .wrap(Math.min(100, yargs.terminalWidth()))
+  .wrap(null)
```
Comment (alxhotel) [inline bin/cmd.js:115]: Maybe don't wrap if `argv.json` is set to true

File: bin/cmd.js [hunk 2]
```diff
@@ -386,22 +387,36 @@ async function runDownload (torrentId) {
     }
 
     // if no index specified, use largest file
-    const index = (typeof argv.select === 'number')
+    argv.select = argv.select = (typeof argv.select === 'number')
```
Comment (alxhotel) [inline bin/cmd.js:390]: Why is this double assignment necessary ?

File: bin/cmd.js [hunk 3]
```diff
@@ -386,22 +387,36 @@ async function runDownload (torrentId) {
...
+      process.stdout.write(JSON.stringify(
+        {
+          href: href,
+          argv: argv,
+          files: torrent.files.map((file, i) => file.name)
+        },
+        null,
+        2
+      ))
+      argv.json = argv.json = false // making sure that it prints only once
```
Comment (alxhotel) [inline bin/cmd.js:417]: Why is this double assignment necessary ?

**Review by alxhotel — CHANGES_REQUESTED** [review]

**Comment by noctuid** [comment]
@OsmanAltun Can you comment the reason for the double assignment? Do you have time to update this PR?

**Pull request closed without merging**